### DISCUSSION TITLE
Skipping build for macOS because /Users/builder/clone/macos runner does not exist.

### DISCUSSION THREAD

**Discussion by poka-IT** [codemagic-ci-cd/codemagic-docs#2693] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Flutter (Dart)

### Steps to reproduce

1 Go to build for macos
2 Read the error

### Expected results

On a basic macos worflow, I have this error for this buid: https://codemagic.io/app/66cb59b74a0e9148a7e34651/build/66cb5c19e8df3a9a879d5805

cannot build for macos a simple CLI Dart tool

### Actual results

On the "building macOS" step
Skipping build for macOS because /Users/builder/clone/macos runner does not exist.

### Build id (optional)

66cb5c19e8df3a9a879d5805

**Comment by dtrdic** [comment]
hey @poka-IT if your project is a mono repo, you need to select the path to your app
![Screenshot 2024-08-27 at 09 29 05](https://github.com/user-attachments/assets/f32915eb-fa59-4194-8053-16cf7e492e29)

**Reply by poka-IT** [reply]
I'm sorry but there is no special path for my repo, it's a simple CLI Dart tool, this form doesn't show me correct paths, it should be ., so the problem is still there.

**Reply by dtrdic** [reply]
hi @poka-IT i can see this error message: 
`build for macOS because /Users/builder/clone/api/bff_generated/macos runner does not exist.` which implies the macos runner is not in the root of your app but at `api/bff_generated`.
Have you tried to select that path in the dropdown and rerun the build? 
If you don't see any project path on the list please click on the refresh button next to the dropdown list

**Reply by poka-IT** [reply]
Yes I tried every possible path just to test, but always same problem, and should be root.

**Reply by dtrdic** [reply]
then that means that the `macos` directory is missing from your Flutter project. Can you try locally to run `flutter create .` which should add the necessary macOS (and other missing platform folders) to your Flutter project

**Reply by poka-IT** [reply]
I understand the problem. This is not a Flutter projet. This just a Dart project. A CLI tool, no Flutter SDK needed.
I'm just looking for a CI solution to build this CLI tool with only Dart SDK, not Flutter SDK, for Linux, Mac and Windows.

**Reply by dtrdic** [reply]
@poka-IT then you will need to switch to .yaml configuration. 
Workflow editor is intended only for Flutter projects. Please add the codemagic.yaml file in the root of your project and then click on the `Switch to YAML configuration` button in the Workflow editor
![image](https://github.com/user-attachments/assets/3f12056d-890d-45a3-8743-b540bc3e026f)

Let me also share a [sample yaml configuration](https://github.com/codemagic-ci-cd/codemagic-sample-projects/blob/main/flutter/flutter-macos-demo-project/codemagic.yaml) to use as a reference

**Comment by dtrdic** [comment]
I'm tentatively closing the discussion since we haven't heard back from you. You can open the discussion again by replying to this message or opening a new discussion.
